### PR TITLE
Add NoOpTracerProvider test for starlette

### LINKED ISSUE #963: Audit and test opentelemetry-instrumentation-starlette

### PR DESCRIPTION
**Pull request by mspillane**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #963

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

tox -e test-instrumentation-starlette

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x]  Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by xrmx** [comment]
@mspillane please sign the CLA

File: instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py [hunk 1]
```diff
@@ -346,6 +346,21 @@ def test_uninstrument(self):
...
+    def test_no_op_tracer_provider(self):
+        self._client.get("/foobar")
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 3)
+
+        self.memory_exporter.clear()
+        self._instrumentor.uninstrument()
+
+        tracer_provider = NoOpTracerProvider()
+        self._instrumentor.instrument(tracer_provider=tracer_provider)
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py:358]: Might want to reset the status of `self._instrumentor` to the original before each test.

Comment (mspillane) [reply instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py:359]: @lzchen I have updated that test to reset the self._instrumentor to the original as requested.

Comment (xrmx) [reply instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py:367]: @mspillane I think he meant for each test and in order to do so you can override the `setUp` method on the testcase

Comment (xrmx) [reply instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py:367]: @mspillane sort of, you uninistrumented it on line 363 so you need that line

**Comment by lzchen** [comment]
@mspillane 

Looks good! Please rebase and allow for maintainer edits so we can resolve conflicts for you.

**Comment by emdneto** [comment]
Closing as fixed in #2654

**Pull request closed without merging**